### PR TITLE
Upload when charging

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
See #465

### CODE REVIEW

**Comment by jancborchardt** [comment]
@tobiasKaminsky can you post a screenshot of how this looks like so I can design-review it?

**Comment by tobiasKaminsky** [comment]
upload_1
upload_2

Currently there is only one "upload when charging", but two "upload on wifi"...
I do not know what my intention was...

**Comment by masensio** [comment]
Hi @tobiasKaminsky!

I'm reviewing this PR. The branch is outdated, but don't worry. I will updated it. 
I'll cross my fingers when resolve conflicts

File: res/xml/preferences.xml [hunk 1]
```diff
@@ -18,15 +18,29 @@
...
 	<PreferenceCategory android:title="@string/prefs_category_security">
-	    <android.preference.CheckBoxPreference android:title="@string/prefs_passcode" android:key="set_pincode" />
+	    <android.preference.CheckBoxPreference
+			android:title="@string/prefs_passcode"
+			android:key="set_pincode" />
 	</PreferenceCategory>
 
-    <PreferenceCategory android:title="@string/prefs_category_instant_uploading" android:key="instant_uploading_category">
-         <com.owncloud.android.ui.CheckBoxPreferenceWithLongTitle android:key="instant_uploading"
+    <PreferenceCategory android:title="@string/prefs_category_instant_uploading"
```
Comment (masensio) [inline res/xml/preferences.xml]: Maybe, it is necessary to review the order in the preferences.xml

Comment (tobiasKaminsky) [inline res/xml/preferences.xml]: @jancborchardt can review the order?

**Comment by tobiasKaminsky** [comment]
@jancborchardt Please have a look here

File: res/xml/preferences.xml (hunk 1 above)
Comment (jancborchardt) [inline res/xml/preferences.xml]: Which order is this actually?

**Comment by jancborchardt** [comment]
Looks good! Just a few details:
* can you change the text to »Upload folder«?
* can you change »WiFi« to »wifi«?
* I assume the folder, wifi and charging options are greyed out when the main instant upload option is not enabled?

**Comment by tobiasKaminsky** [comment]
ref 1:"Upload folder"? We do not support this currently, or?
This is all about the option, that the upload only starts if the phone is charging.

2: I will change
3: As far as I remember, they are not visible, if the main option is disabled.

**Comment by jancborchardt** [comment]
@tobiasKaminsky right, if we call it »Upload folder« it might get confusing. »Upload path« is better, but with »path« in lowercase.

**Comment by tobiasKaminsky** [comment]
instantupload1

Should "when charging" be independent for picture and video as "on wifi"?

**Comment by tobiasKaminsky** [comment]
Updated text:
instantupload2

@jancborchardt Remaining question is, if "when charging" should be an independent setting for picture and video? This could be a good idea as videos are very big and can drain the battery very quick...

**Comment by jancborchardt** [comment]
@tobiasKaminsky yeah, I would make the setting independent. Right now on the very top it’s kind of confusing anyway.

**Comment by tobiasKaminsky** [comment]
instantupload3

**Comment by jancborchardt** [comment]
Yeah, looks good!

**Pull request closed without merging**